test/react/first-render-only/simple-3.js

```diff
@@ -0,0 +1,40 @@
+var React = require('react');
+// the JSX transform converts to React, so we need to add it back in
+this['React'] = React;
+var checkRef = false;
+
+function App(props) {
+  function fn() {
+    // should not be here
```

Comment:
Do we check this anyhow?

Reply:
Not yet, this test is more for my follow up PR. I left it in a comment for now saying just that.